src/loaders/compute_tools/microtrait/microtrait.py

```diff
@@ -67,7 +67,10 @@ def _run_microtrait(genome_id: str, fna_file: Path, genome_dir: Path, debug: boo
 
     microtrait_result = _get_r_list_element(r_result, 'microtrait_result')
 
-    trait_counts = _get_r_list_element(microtrait_result, TRAIT_COUNTS_ATGRANULARITY)
+    try:
+        trait_counts = _get_r_list_element(microtrait_result, TRAIT_COUNTS_ATGRANULARITY)
+    except Exception as e:
```

Comment:
This is too broad - it'll catch exceptions that should be thrown. What should happen is that there should be a test that checks whether the key exists or not, and if not, adds some sort of information to the output file (or an alternate file if that'll screw up the output file) that tells the parser (and anyone that reads the output file) that microtrait didn't work. The parser should then check that file and log that it's skipping that result.

IOW, there needs to be a positive identification that a known error mode occurred that the parser can key off of, and not just missing data.

Reply:
Updated!

Reply:
A `ValueError` is still too broad - lots of things throw that. Instead of catching an error, test to see if the key is in the pandas columns set (maybe `index` returns -1 if it isn't? not sure) and don't catch errors at all - or throw a custom error from `_get_r_list_element` that is only thrown in this context and catch that

Reply:
return exist boolean and write to fatal_error.json if exist boolean is False

Reply:
:+1: